### PR TITLE
Enhance Header Parsing and Error Handling in HTTP Driver

### PR DESCRIPTION
**Pull request by dmikey**
This pull request introduces several improvements and fixes to the HTTP driver, specifically targeting the parsing and handling of headers from JSON input. The changes include enhanced error handling, robust test coverage, and avoiding unnecessary dependencies.

* Added graceful error handling of header parsing to avoid unwrap errors.
* Added tests to cover various scenarios
    * Added tests for valid headers to ensure correct parsing.
    * Added tests for invalid JSON structures to verify error handling.
    * Added tests for invalid header names and values to ensure appropriate error reporting.
* Derived PartialEq for HttpErrorKind to facilitate comparison in tests

These changes significantly improve the robustness and reliability of the HTTP driver when parsing headers from JSON input. By enhancing error handling and increasing test coverage, we ensure that the driver can gracefully handle various edge cases and invalid inputs

### CODE REVIEW

File: crates/blockless-drivers/src/http_driver/reqwest_driver.rs [hunk 1]
```diff
@@ -58,22 +66,35 @@ pub(crate) async fn http_req(url: &str, opts: &str) -> Result<(u32, i32), HttpEr
...
+            let header_name = match reqwest::header::HeaderName::from_bytes(key.as_bytes()) {
+                Ok(name) => name,
+                Err(_) => return Err(HttpErrorKind::HeadersValidationError),
+            };
+
+            // Handle possible errors from from_str
+            let header_value = match reqwest::header::HeaderValue::from_str(value.as_str().unwrap_or_default()) {
+                Ok(value) => value,
+                Err(_) => return Err(HttpErrorKind::HeadersValidationError),
+            };
```
Comment (zees-dev) [inline crates/blockless-drivers/src/http_driver/reqwest_driver.rs:95]: nit: simplification:
```suggestion
        let header_name = reqwest::header::HeaderName::from_bytes(key.as_bytes())
            .map_err(|_| HttpErrorKind::HeadersValidationError)?;

        let header_value = reqwest::header::HeaderValue::from_str(value.as_str().unwrap_or_default())
            .map_err(|_| HttpErrorKind::HeadersValidationError)?;
```

Comment (dmikey) [reply crates/blockless-drivers/src/http_driver/reqwest_driver.rs:95]: Thanks! I like this! I clean it up!

File: crates/blockless-drivers/src/error.rs [hunk 2]
```diff
@@ -33,7 +33,7 @@ impl std::fmt::Display for ErrorKind {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub enum HttpErrorKind {
     InvalidDriver,
     InvalidHandle,
```
Comment (zees-dev) [inline crates/blockless-drivers/src/error.rs:39]: nit:
If `PartialEq` is only required for dev/tests, then does not need to be derived for non-test envs.

```suggestion
#[derive(Debug)]
#[cfg_attr(test, derive(PartialEq))]
pub enum HttpErrorKind {
    InvalidDriver,
    InvalidHandle,
```

**Review by zees-dev — APPROVED** [review]
LGTM 👍 

To pass CI you may need to run `cargo fmt --all`

**Review by Joinhack — APPROVED** [review]

**Comment by Joinhack** [comment]
Don't worry, I format it.
